src/NSwag.CodeGeneration.TypeScript/Templates/Client.ProcessResponse.liquid

```diff
@@ -3,7 +3,7 @@
 let _mappings: { source: any, target: any }[] = [];
 {% endif -%}
 {% for response in operation.Responses -%}
-if (status === {{ response.StatusCode }}{% if response.CheckChunkedStatusCode %} || status === 206{% endif %}) {
+{% if response.StatusCodeIsInt %}if (status === {{ response.StatusCode }}{% if response.CheckChunkedStatusCode %} || status === 206{% endif %}){% else %}if (`${status}`.match('^{{ response.StatusCode }}$'.replace(/x/gi, '\\d'))){% endif %} {
```

Comment:
Same as in CS case, using external input as regex should probably avoided. Maybe use something like `(`${status/100}` + 'xx') === response.StatusCode.toLower()`. (pseudo code)